File: Diagrammatize.Design.Pattern/code/python/Flyweight/Sample/BigString.py

from BigCharFactory import BigCharFactory

class BigString:

    def __init__(self, string):
        bigchars = [None]*len(string)
        factory = BigCharFactory.getInstance()
        for i in range(len(string)):
            bigchars[i] = factory.getBigChar(string[i])
        self.bigchars = bigchars

    def print(self):
        for i in range(len(self.bigchars)):
            self.bigchars[i].print()

File: Diagrammatize.Design.Pattern/code/cxx/Flyweight/Sample/BigString.cpp
#include "BigString.hpp"
#include "BigCharFactory.hpp"

BigString::BigString(const std::string& string) {
    bigchars.resize(string.length());
    auto& factory = BigCharFactory::getInstance();
    for (int i = 0; i < bigchars.size(); i++) {
        bigchars[i] = factory.getBigChar(string[i]);
    }
}

void BigString::print() {
    for (int i = 0; i < bigchars.size(); i++) {
        bigchars[i]->print();
    }
}
